functions/packages.bash

```diff
@@ -798,3 +798,160 @@ setup_evcc() {
+
+    # Disable the ESPHome Dashboard service
+    echo -n "$(timestamp) [openHABian] Disabling the ESPHome Dashboard service..."
+    systemctl disable esphome-dashboard.service
+    check_success "Failed to disable ESPHome Dashboard service."
+
+    # Remove the ESPHome Dashboard service file
+    echo -n "$(timestamp) [openHABian] Removing the ESPHome Dashboard systemd service file..."
+    rm -f /etc/systemd/system/esphome-dashboard.service
+    check_success "Failed to remove systemd service file."
+
+    # Reload systemd daemon
+    echo -n "$(timestamp) [openHABian] Reloading systemd daemon..."
+    systemctl daemon-reload
+    check_success "Failed to reload systemd daemon."
+
+    # Remove the ESPHome installation directory
+    ESPHOME_DIR="/opt/esphomedashboard"
+    echo -n "$(timestamp) [openHABian] Removing ESPHome directory at $ESPHOME_DIR..."
+    rm -rf "$ESPHOME_DIR"
+    check_success "Failed to remove $ESPHOME_DIR."
+
+    # Final cleanup 
+    echo -n "$(timestamp) [openHABian] Uninstallation complete!"
+    echo -n "$(timestamp) [openHABian] ESPHome Dashboard has been completely removed from the system."
+    return;
+  fi
+
+  if [[ $1 != "install" ]]; then 
+    if [[ -n $INTERACTIVE ]]; then
+      whiptail --title "ESPhome dashboard installation" --msgbox "$installText" 8 80
+    fi
+    echo -n "$(timestamp) [openHABian] Starting ESPHome Dashboard setup..."
+
+    # Function to check the success of the last executed command
+    check_success() {
+        if [ $? -ne 0 ]; then
+            echo -n "$(timestamp) [openHABian] Error: $1"
+            exit 1
+        fi
+    }
+
+    # Ensure the script is run with sudo
+    if [ "$EUID" -ne 0 ]; then
+        echo -n "$(timestamp) [openHABian] Please run this script as root or with sudo."
+        exit 1
+    fi
+
+    # Update system packages
+    echo -n "$(timestamp) [openHABian] Updating system packages..."
+    apt update && apt upgrade -y
+    check_success "Failed to update system packages."
+
+    # Install Python 3 and pip
+    echo -n "$(timestamp) [openHABian] Installing Python 3 and pip..."
+    apt install -y python3 python3-pip python3-venv
+    check_success "Failed to install Python 3 and pip."
+
+    # Create the /opt/esphomedashboard directory and set permissions
+    ESPHOME_DIR="/opt/esphomedashboard"
+    echo -n "$(timestamp) [openHABian] Creating directory at $ESPHOME_DIR..."
+    mkdir -p "$ESPHOME_DIR/config"
+    check_success "Failed to create $ESPHOME_DIR and its config directory."
+   
+    # Set ownership to the current non-root user
+    USER=$(logname)
+    chown -R "$USER:$USER" "$ESPHOME_DIR"
+    check_success "Failed to set ownership of $ESPHOME_DIR to $USER."
+
+    # Switch to the ESPHome directory and set up a virtual environment
+    echo -n "$(timestamp) [openHABian] Setting up a virtual environment in $ESPHOME_DIR..."
+    cd "$ESPHOME_DIR" || exit
+    sudo -u "$USER" python3 -m venv venv
```

Comment:
Rather than `cd` it would be better to simply provide the full path when calling `venv` and it should create it at that path

Reply:
here i need some help from you @ecdye

Reply:
Something like:
``` bash
sudo -u "$USER" python3 -m venv "$ESPHOME_DIR/venv"
```
should work

Reply:
solved